### PR TITLE
Add RFC for modulo operator

### PR DESCRIPTION
**Pull request by mfelsche**
and connected floored division for all integer types.

[Rendered](https://github.com/ponylang/rfcs/blob/mod-operator/text/0000-modulo-operator.md)

### CODE REVIEW

File: text/0000-modulo-operator.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+- Feature Name: modulo_operator
+- Start Date: 2018-11-09
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+Add a modulo operator `%%` for integer types that is defined as the remainder after floor division. Most importantly the result always has the sign of the Divisor, in contrast to `rem` (`%` operator), whose results always have the result of the dividend.
```
Comment (jemc) [inline text/0000-modulo-operator.md]: Small typo:

> whose results always have the result of the dividend

Should be:

> whose results always have the *sign* of the dividend

**Comment by aturley** [comment]
@malthe are you ready for final comment period on this?

**Comment by malthe** [comment]
I think you mean @mfelsche

**Comment by mfelsche** [comment]
I am, lets do this

**Comment by jemc** [comment]
For reference, see also:
https://github.com/ponylang/ponyc/pull/2888
https://github.com/ponylang/pony-tutorial/pull/323

(in which `%` was renamed as `rem` from previously being named `mod`)

**Comment by SeanTAllen** [comment]
I think this RFC needs to define what happens for floating point. For example, `rem` exists as a method on F64. Will `mod` exist on F64? Will `%%` work on it?

See https://playground.ponylang.io/?gist=0f84691aa4de5367ff7cb7021a9a8cec to see that `rem` works on F64 now.

I think this RFC makes sense but we would 

- want floating point and integer to have the same operations available
- need to define for `%%` and `mod` what happens with floating point operations.

From sync, Sylvan suggests that for floating point we should use divisor.

**Comment by mfelsche** [comment]
Yes, this RFC totally needs to also speak about floats, i totally overlooked them also having a rem method. Gonna add that within the next days

**Comment by aturley** [comment]
Removing "final comment period" tag so @mfelsche can add stuff about floats.

**Comment by mfelsche** [comment]
Stuff about floats has been added, namely: mod and fld (floored division) will also be added to floating point types with mod being the remainder after division with the same aign as the divisor and without truncation (except normal IEEE floating point truncation).

I think, thus that this is read for final comment period.

**Comment by jemc** [comment]
@mfelsche - did you forget to push the commit?

**Comment by mfelsche** [comment]
Yep, sorry, gonna push tonight. Have been in a rush...

**Comment by mfelsche** [comment]
pushed.

**Comment by jemc** [comment]
:+1:

**Comment by aturley** [comment]
@SeanTAllen do you have any input on this?

**Comment by skull-squadron** [comment]
Will there be `%%?` and `%%~` to stay consistent?

**Comment by mfelsche** [comment]
Yes, i will add those. Gonna update this RFC to make this clear.

**Comment by jemc** [comment]
Thanks @mfelsche!

**Pull request merged**